src/models/applications.js

```diff
@@ -319,14 +319,32 @@ var FlatpakApplicationsModel = GObject.registerClass({
         if (component.get_name())
             appdata.name = component.get_name();
 
-        if (component.get_developer_name())
-            appdata.author = component.get_developer_name();
+        try {
+            /* AppStream 1.0.0+ */
+            const developer = component.get_developer();
+            if (developer && developer.get_name())
+                appdata.author = developer.get_name();
+        } catch (err) {
+            /* AppStream up to 0.16 */
+            if (component.get_developer_name())
+                appdata.author = component.get_developer_name();
+        }
 
         const launchable = component.get_launchable(AppStream.LaunchableKind.DESKTOP_ID);
         if (launchable && launchable.get_entries())
             [appdata.launchable] = launchable.get_entries();
 
-        const [release] = component.get_releases();
+        let release = [];
+        try {
+            /* AppStream 1.0.0+ */
+            const releaselist = component.get_releases_plain();
+            if (!releaselist)
+                return appdata;
+            [release] = releaselist.get_entries();
+        } catch (err) {
+            /* AppStream up to 0.16 */
+            [release] = component.get_releases();
+        }
```

Comment:
Thanks for working on this, can you please modify this to assume it's running with AppStream 1.0.0+  only? (no try-except block, no back-compatibility).

I will merge it once the gnome runtime bumps AppStream to that version and, in the meantime, you should contact whoever is distributing flatseal outside of flathub to patch this downstream.

Version guessing is not something I want upstream.